README: add download section

aparcar opened the pull request:
The *Firmware Selector* was build to simplify users finding suitable
images for their devices. Let's give it more visibility so unleashing
your home router becomes an easier thing.

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar committed:
README: add download section

The *Firmware Selector* was built to simplify users finding suitable
images for their devices. Let's give it more visibility so unleashing
your home router becomes an easier thing.

As an alternative for more advanced user, provide the directl link to
the wiki download section.

Signed-off-by: Paul Spooren <mail@aparcar.org>
[ fix typo in commit description, add link to Wiki download page ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

neheb commented:
/s/build/built

BKPepe commented on README.md:
```diff
@@ -11,6 +11,18 @@ full customization, to use the device in ways never envisioned.
+
+* [OpenWrt Firmware Selector](https://firmware-selector.openwrt.org/)
+
+If your device is supported please follow the **Info** link to see install
```
```suggestion
If your device is supported, please follow the **Info** link to see the install
```

BKPepe commented on README.md:
```diff
@@ -11,6 +11,18 @@ full customization, to use the device in ways never envisioned.
+
+Built firmware images are available for many architectures and come with a
+package selection to be used as WiFi home router. To quickly find a factory
+image usable to migrate from a vendor stock firmware to OpenWrt try the
```
```suggestion
image usable to migrate from a vendor stock firmware to OpenWrt, try the
```

aparcar commented:
@adschm do you have an opinion on this?

adschm commented:
I'd personally not highlight just the firmware selector, but would link a general downloads landing page here.

This page then of course could advertise the firmware selector next to the other options to download images.

Apart from that detail, I actually (generally) expect to find download links for built stuff on a GitHub repo README, so I'm quite surprised we didn't offer that until now.

aparcar commented:
@adschm do you mean https://downloads.openwrt.org/ or https://openwrt.org/downloads?

adschm commented:
> @adschm do you mean https://downloads.openwrt.org/ or https://openwrt.org/downloads?

I'm not really sure, but it should be one of them.

Ansuel commented:
> https://openwrt.org/downloads?

@adschm @aparcar @neheb I would use this as it looks more appealing... Why this still is not merged?

neheb commented:
for one the description has a typo, build vs. built.

neheb commented:
anyway, ACK on this commit.

Ansuel commented:
@aparcar can you fix the typo so we can merge this?

aparcar commented:
@Ansuel what missing typos?

Ansuel commented:
@aparcar 

The Firmware Selector was **build** to simplify users finding suitable
images for their devices. Let's give it more visibility so unleashing
your home router becomes an easier thing.

Should be built i think

aparcar commented:
@adschm I'd at least link both, the firmware selector and the general download page. From my understanding of usability one should go straight to a website with download after clicking on something called "downloads". Our current "downloads" page contains many many sub pages and a raw folder structure with targets/subtargets.

Ansuel committed:
CI: tools: compile tools with ccache support for tools container

Enable ccache support for tools container, useful to speedup other
workflow even more.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

Ansuel committed:
CI: kernel: use ccache to speedup workflow

Use ccache to speedup kernel compilation.
Ccache dir is cached across each build test. To refresh ccache directory
we generate an hash of the kernel include files, that includes the
kernel versions of every kernel supported and the kernel compile
includes.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
